hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/delegation/DelegationConstants.java

```diff
@@ -160,6 +160,18 @@ public final class DelegationConstants {
    */
   public static final String STS_STANDARD = "sts.amazonaws.com";
 
+  /**
+   * The format of the STS Endpoint
+   */
+  public static final String STS_ENDPOINT_URI_PATTERN = "^sts\\..*\\.amazonaws\\.com$";
```

Comment:
your pattern will need to handle china as well.

Reply:
Done ! Please do check the lastest changes and the additional test that was added.